provider/vsphere/environ_policy.go

```diff
@@ -71,6 +74,24 @@ func (env *sessionEnviron) checkDatastore(ctx context.ProviderCallContext, datas
 	return errors.NotFoundf("datastore %q", name)
 }
 
+func (env *sessionEnviron) checkExtendPermissions(ctx context.ProviderCallContext, rootDisk *uint64) error {
+	if rootDisk == nil || *rootDisk == 0 {
+		return nil
+	}
+	// If we're going to need to resize the root disk we need to have
+	// the System.Read privilege on the root level folder - this seems
+	// to be because the extend disk task doesn't get created with a
+	// target, so the root-level permissions are applied.
+	ok, err := env.client.UserHasRootLevelPrivilege(env.ctx, "System.Read")
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if !ok {
+		return errors.Errorf("the System.Read privilege is required at the root level to extend disks - please grant the ReadOnly role to this model's user")
```

Comment:
can we include the user name in the message?

Reply:
Yeah, good call - I'll see if that's accessible here.

Reply:
It was!